packages/api/models/rules.js

```diff
@@ -206,67 +204,38 @@ class Rule extends Manager {
    */
   async addKinesisEventSource(item, lambda) {
     // use the existing event source mapping if it already exists and is enabled
-    const listParams = { FunctionName: lambda.name };
-    const listData = await aws.lambda(listParams).listEventSourceMappings().promise();
+    const listParams = {
+      FunctionName: lambda.name,
+      EventSourceArn: item.rule.value
+    };
+    const listData = await aws.lambda().listEventSourceMappings(listParams).promise();
```

Comment:
This was a bug. It was returning all event mappings (for the account i think?) and I think we could get into a situation where it would not return all of them and we'd try to create it again and it would fail. This seems to have fixed the issue.